sound/soc/sof/pcm.c

```diff
@@ -756,6 +756,12 @@ static int sof_pcm_probe(struct snd_soc_component *component)
 	struct snd_sof_pdata *plat_data = sdev->pdata;
 	const char *tplg_filename;
 	int ret;
+	struct platform_device *pdev = plat_data->pdev_mach;
+	struct snd_soc_acpi_mach *mach = pdev->dev.platform_data;
```

Comment:
I might be wrong but this doesn't look right to me. This is a generic pcm.c file, it has nothing to do with ACPI. And not every SOF implementation has ACPI information. Why not do that in `sof_acpi_probe()`? Any reason why that wouldn't work? And you want to use `dev_get_platdata();`

Reply:
@lyakh this change is not specific for ACPI platforms. For the Intel platforms, we always have this information in the machine driver platform device.  On the other hand, what I am not sure about is if this applies to DT platforms. Something for @dbaluta to confirm.

Reply:
@sathya-nujella @ranj063 this doesn't looks good from Device Tree perspective. If you want to override the tplg_filename in the worst case please create a generic snd_sof_* function that gets implemented by each platform.

This could apply to DT platforms also (overriding the name of the topology file). We use this kind of trick to update the name of the machine driver.

https://github.com/dbaluta/linux/blob/next/sound/soc/sof/pcm.c#L793

To sum up, I'm not against this change if it is made in a generic way so that no ACPI stuff is added to generic pcm.c file.

Reply:
We should do this in sof-pci-dev.c

We can add a new DMI table, to override specific topology files.

if (dmi_check_system(google_alternate_topology_platforms)) {
		sof_pdata->tplg_filename = sof_override_tplg_name;
}
	
we might even be able to avoid this extra variable and directly get the data from the quirk.

Reply:
> We should do this in sof-pci-dev.c
> 
> We can add a new DMI table, to override specific topology files.
> 
> ```c
> if (dmi_check_system(google_alternate_topology_platforms)) {
> 		sof_pdata->tplg_filename = sof_override_tplg_name;
> }
> ```
> 
> we might even be able to avoid this extra variable and directly get the data from the quirk.

Thank you @plbossart . I have removed previous tplg dmi method and added in sof-pci-dev.

Reply:
> @sathya-nujella @ranj063 this doesn't looks good from Device Tree perspective. If you want to override the tplg_filename in the worst case please create a generic snd_sof_* function that gets implemented by each platform.
> 
> This could apply to DT platforms also (overriding the name of the topology file). We use this kind of trick to update the name of the machine driver (will need to find a generic way, when have some time to upstream my patch).
> 
> https://github.com/dbaluta/linux/blob/next/sound/soc/sof/pcm.c#L793
> 
> To sum up, I'm not against this change if it is made in a generic way so that no ACPI stuff is added to generic pcm.c file.

I removed earlier method and changed based on latest inputs. Thanks!